cmd/wsm/list.go

```diff
@@ -0,0 +1,177 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"sort"
+
+	"github.com/charmbracelet/bubbles/spinner"
+	"github.com/charmbracelet/bubbletea"
+	"github.com/charmbracelet/lipgloss"
+	"github.com/spf13/cobra"
+	"github.com/wandb/wsm/pkg/deployer"
+	"github.com/wandb/wsm/pkg/helm"
+	"github.com/wandb/wsm/pkg/utils"
+)
+
+var (
+	listStyle = lipgloss.NewStyle().
+			Foreground(lipgloss.Color("12")).
+			Bold(true)
+
+	itemStyle = lipgloss.NewStyle().
+			Foreground(lipgloss.Color("10")).
+			PaddingLeft(2)
+
+	footerStyle = lipgloss.NewStyle().
+			Foreground(lipgloss.Color("14")).
+			Bold(true)
+)
+
+type model struct {
+	spinner spinner.Model
+}
+
+func (m model) Init() tea.Cmd {
+	return nil
+}
+
+func (m model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	var cmd tea.Cmd
+	m.spinner, cmd = m.spinner.Update(msg)
+	return m, cmd
+}
+
+func (m model) View() string {
+	return m.spinner.View()
+}
+
+// Function to fetch the latest tag from Docker Hub API
+func getLatestWandbTag() (string, error) {
+	url := "https://registry.hub.docker.com/v2/repositories/wandb/controller/tags/"
+	resp, err := http.Get(url)
+	if err != nil {
+		return "", fmt.Errorf("error fetching tags: %v", err)
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", fmt.Errorf("error reading response body: %v", err)
+	}
+
+	// Parse the JSON response
+	var result map[string]interface{}
+	err = json.Unmarshal(body, &result)
+	if err != nil {
+		return "", fmt.Errorf("error unmarshalling JSON: %v", err)
+	}
+
+	// Extract tags and filter out "latest"
+	var tags []string
+	if results, ok := result["results"].([]interface{}); ok {
+		for _, r := range results {
+			if tag, ok := r.(map[string]interface{})["name"].(string); ok && tag != "latest" {
+				tags = append(tags, tag)
+			}
+		}
+	}
+
+	// Sort tags in natural (version) order
+	sort.Strings(tags)
+
+	// If there are not enough tags, return "latest"
+	if len(tags) < 2 {
```

Comment:
I assume if that for some reason there are not tags something went wrong, we should probably return an error instead

Reply:
Returning error if not enough tags present.